### PR TITLE
Feature/fetch project names

### PR DESCRIPTION
**Pull request by Cvos0162**
#### Description
This fetches project names from the backend API on the main page.

projectServices.ts is created, and I guess other relative frontend service calls to the backend will follow the same fashion.

Please create .env file after this pr.
The service call requires API URL for backend api.
The variable name for the URL is called REACT_APP_API_URL

#### Review Checklist
- [x] The requirements on the description have been met
- [ ] The relevant documentation has been updated:
    - [ ] code comments for classes / methods / complex statements
    - [ ] package readme
    - [ ] documentation for high level approaches or product info
- [x] Tests:
    - [ ] new tests written
    - [ ] existing affected tests updated
    - [x] all green and passing
- [ ] Tested by reviewer

### CODE REVIEW

**Review by mliu433 — COMMENTED** [review]
I have a question on the project name part, what if two project has the same name?

**Comment by Cvos0162** [comment]
> I have a question on the project name part, what if two project has the same name?

This should be handled by the backend API,

File: app/src/pages/MainPage.tsx [hunk 1]
```diff
@@ -59,13 +58,19 @@ import {
             
         <div className="container">
             {projectData.map((name, index) => (
+<<<<<<< HEAD
```
Comment (tnzhu) [inline app/src/pages/MainPage.tsx:61]: Merge conflict here

**Review by tnzhu — CHANGES_REQUESTED** [review]

File: app/src/pages/MainPage.tsx (hunk 1 above)
Comment (Cvos0162) [reply app/src/pages/MainPage.tsx:61]: time to destroy my git

Comment (Cvos0162) [reply app/src/pages/MainPage.tsx]: done!

**Review by tnzhu — APPROVED** [review]
LGTM!

**Pull request merged**